### PR TITLE
TBI(ui5-application-writer) disable code assist with Typescript

### PR DESCRIPTION
**Pull request by devinea**
Disable codeAssist when Typescript is also selected. Since Typescript includes it's own code assist config.

### CODE REVIEW

**Review by longieirl — APPROVED** [review]

**Review by tobiasqueck — APPROVED** [review]
Thanks, the change makes a lot of sense. Existing tests are covering the case and the snapshots are updated correctly. So, all good.

File: packages/ui5-application-writer/src/data/index.ts [hunk 1]
```diff
@@ -18,6 +18,10 @@ export function mergeWithDefaults(ui5App: Ui5App): {
     validate(ui5App);
     ui5App.app = mergeApp(ui5App.app);
     ui5App.appOptions = ui5App.appOptions || {};
+    // if typescript and codeAssist is enabled disable codeAssist
+    if (ui5App.appOptions.typescript && ui5App.appOptions.codeAssist) {
```
Comment (IainSAP) [inline packages/ui5-application-writer/src/data/index.ts:22]: Should we document this behaviour via the types ts docs: 
https://github.com/SAP/open-ux-tools/blob/main/packages/ui5-application-writer/src/types.ts#L57

**Review by IainSAP — APPROVED** [review]
Thx @devinea 

- Tests updated as expected
- No behavioural changes

File: packages/ui5-application-writer/src/data/index.ts (hunk 1 above)
Comment (IainSAP) [reply packages/ui5-application-writer/src/data/index.ts:22]: What if I set `codeAssist` to false and 'typescript' to true? I assume we still get code assist...

Comment (devinea) [reply packages/ui5-application-writer/src/data/index.ts:22]: Yes that would be the same result as both being true. You get the Typescript code assist enabled.

Comment (devinea) [reply packages/ui5-application-writer/src/data/index.ts:22]: Updated in https://github.com/SAP/open-ux-tools/pull/773/commits/daff85d11c822dec010b2b6ef3f006356d0472c4

**Review by IainSAP — APPROVED** [review]
Thx for the doc update, helps to clarify behaviour.

**Review by tobiasqueck — APPROVED** [review]
Thanks for updating the comments.

**Review by longieirl — APPROVED** [review]

**Comment by devinea** [comment]
Thanks all

**Pull request merged**